src/input/compact/InputSettlerCompact.sol

```diff
@@ -54,10 +54,26 @@ contract InputSettlerCompact is InputSettlerPurchase, IInputSettlerCompact {
 
     constructor(
         address compact
-    ) EIP712("OIFCompact", "1") {
+    ) EIP712(_domainName(), _domainVersion()) {
         COMPACT = TheCompact(compact);
     }
 
+    /**
+     * @notice Returns the domain name of the EIP712 signature.
+     * @return name The domain name.
+     */
+    function _domainName() internal pure override returns (string memory) {
```

Comment:
Note that if this is `pure` any overrides must also be `pure`. It might be best to make this `view` to be less restrictive. In that case we should document that it's only read in the constructor and that the value is cached.